Simplified/NYPLBookState.swift

```diff
@@ -0,0 +1,77 @@
+import Foundation
+
+let DownloadingKey = "downloading"
+let DownloadFailedKey = "download-failed"
+let DownloadNeededKey = "download-needed"
+let DownloadSuccessfulKey = "download-successful"
+let UnregisteredKey = "unregistered"
+let HoldingKey = "holding"
+let UsedKey = "used"
+let UnsupportedKey = "unsupported"
+
+@objc public enum NYPLBookState : Int {
+  case Unregistered = 0
+  case DownloadNeeded = 1
+  case Downloading
+  case DownloadFailed
+  case DownloadSuccessful
+  case Holding
+  case Used
+  case Unsupported
+    
+  init?(_ stringValue: String) {
+    switch stringValue {
+      case DownloadingKey:
+        self = .Downloading
+      case DownloadFailedKey:
+        self = .DownloadFailed
+      case DownloadNeededKey:
+        self = .DownloadNeeded
+      case DownloadSuccessfulKey:
+        self = .DownloadSuccessful
+      case UnregisteredKey:
+        self = .Unregistered
+      case HoldingKey:
+        self = .Holding
+      case UsedKey:
+        self = .Used
+      case UnsupportedKey:
+        self = .Unsupported
+      default:
```

Comment:
do we need the default `nil` case if we have Unsupported? If not, the initializer doesn't need to be optional, which would be nice.  But I am just wondering.

Reply:
I believe "Unsupported" is an actual book state, not a statement about the enum itself.

Reply:
gotcha, never mind then.